java-checks/src/main/java/org/sonar/java/checks/EmptyStatementUsageCheck.java

```diff
@@ -40,17 +39,21 @@
 
   @Override
   public void visitNode(Tree tree) {
-    if (usedForEmptyEnum(tree)) {
+    if (usedForEmptyEnum(tree) || uniqueStatementOfLoop(tree)) {
       return;
     }
     reportIssue(tree, "Remove this empty statement.");
   }
 
   private static boolean usedForEmptyEnum(Tree tree) {
     Tree parent = tree.parent();
-    if(parent.is(Tree.Kind.ENUM)) {
+    if (parent.is(Tree.Kind.ENUM)) {
       return ((ClassTree) parent).members().indexOf(tree) == 0;
     }
     return false;
   }
+
+  private boolean uniqueStatementOfLoop(Tree tree) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "uniqueStatementOfLoop" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&branch=SONARJAVA-2661&issues=AWH7-fhja5Ltb7kATz4Y&open=AWH7-fhja5Ltb7kATz4Y)

[ISSUE_KEY]: AWH7-fhja5Ltb7kATz4Y